From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Keeley Hammond <vertedinde@electronjs.org>
Date: Tue, 21 Feb 2023 16:09:35 -0800
Subject: build: use Visual Studio 2022 v17.5.0

CL: https://chromium-review.googlesource.com/c/chromium/src/+/4072083
When updating Chromium's Windows toolchain, VS is only checking for 17.0.
Also use 17.5 if it's available.

Patch_Filename: use_visual_studio_2022_v17.5.0

diff --git a/build/vs_toolchain.py b/build/vs_toolchain.py
index a420dad636df4e954bfd5b074fb6171084c530d7..5bcb965f1aa91416d74b61160e64ec2b00e7323a 100755
--- a/build/vs_toolchain.py
+++ b/build/vs_toolchain.py
@@ -47,7 +47,8 @@ json_data_file = os.path.join(script_dir, 'win_toolchain.json')
 # The first version is assumed by this script to be the one that is packaged,
 # which makes a difference for the arm64 runtime.
 MSVS_VERSIONS = collections.OrderedDict([
-    ('2022', '17.0'),  # Default and packaged version of Visual Studio.
+    ('2022', '17.5'),  # Default and packaged version of Visual Studio.
+    # ('2022', '17.0'),
     ('2019', '16.0'),
     ('2017', '15.0'),
 ])
@@ -183,10 +184,10 @@ def GetVisualStudioVersion():
       available_versions.append(version)
       break
     # Detecting VS under possible paths.
-    if version >= '2022':
-      program_files_path_variable = '%ProgramFiles%'
-    else:
-      program_files_path_variable = '%ProgramFiles(x86)%'
+    # if version >= '2022':
+    #   program_files_path_variable = '%ProgramFiles%'
+    # else:
+    program_files_path_variable = '%ProgramFiles(x86)%'
     path = os.path.expandvars(program_files_path_variable +
                               '/Microsoft Visual Studio/%s' % version)
     if path and any(
